// kovaaks-api-client/dist/client.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KovaaksApiClient = exports.KovaaksApiError = void 0;
const axios_1 = __importDefault(require("axios"));
/**
 * Custom error class for API-related errors
 */
class KovaaksApiError extends Error {
    constructor(message, status, response) {
        super(message);
        this.status = status;
        this.response = response;
        this.name = 'KovaaksApiError';
    }
}
exports.KovaaksApiError = KovaaksApiError;
/**
 * Kovaaks API Client
 *
 * A TypeScript client for interacting with the Kovaaks API.
 * Provides access to leaderboards, user profiles, scenarios, playlists, and benchmarks.
 */
class KovaaksApiClient {
    constructor(options = {}) {
        const baseURL = options.baseUrl || 'https://kovaaks.com';
        this.client = axios_1.default.create({
            baseURL,
            timeout: options.timeout || 30000,
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
            },
            ...options.customRequestConfig
        });
        if (options.authToken) {
            this.setAuthToken(options.authToken);
        }
        // Add response interceptor for error handling
        this.client.interceptors.response.use((response) => response, (error) => {
            var _a, _b, _c, _d;
            const message = ((_b = (_a = error.response) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.error) || error.message || 'Unknown error';
            const status = ((_c = error.response) === null || _c === void 0 ? void 0 : _c.status) || 500;
            const response = (_d = error.response) === null || _d === void 0 ? void 0 : _d.data;
            throw new KovaaksApiError(message, status, response);
        });
    }
    /**
     * Set the authentication token for future requests
     */
    setAuthToken(token) {
        this.authToken = token;
        this.client.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    }
    /**
     * Clear the authentication token and log out
     */
    logout() {
        this.authToken = undefined;
        delete this.client.defaults.headers.common['Authorization'];
    }
    /**
     * Helper method to build URL with query parameters
     */
    buildUrl(path, params = {}) {
        const url = new URL(path, this.client.defaults.baseURL);
        Object.entries(params).forEach(([key, value]) => {
            if (value !== undefined && value !== null) {
                if (Array.isArray(value)) {
                    value.forEach(v => url.searchParams.append(`${key}[]`, String(v)));
                }
                else {
                    url.searchParams.append(key, String(value));
                }
            }
        });
        return url.toString();
    }
    /**
     * Make an HTTP request
     */
    async request(config) {
        try {
            const response = await this.client.request(config);
            return response.data;
        }
        catch (error) {
            if (error instanceof KovaaksApiError) {
                throw error;
            }
            throw new KovaaksApiError(error.message || 'Request failed', 500);
        }
    }
    async searchScenariosByName(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/scenario/popular', {
                scenarioNameSearch: params.scenarioName,
                page: params.page,
                max: params.max
            })
        });
    }
    async getBenchmarkProgress(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/benchmarks/player-progress-rank-benchmark', {
                benchmarkId: params.benchmarkId,
                steamId: params.steamId,
                page: params.page,
                max: params.max
            })
        });
    }
    async getGlobalLeaderboard(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/leaderboard/global/scores', params)
        });
    }
    async getBenchmarkProgressForUsername(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl(`/webapp-backend/benchmarks/player-progress-rank`, {
                username: params.username,
                page: params.page,
                max: params.max
            })
        });
    }
    async getPlaylistsByUser(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/user/playlist/creator', {
                username: params.username,
                page: params.page,
                max: params.max
            })
        });
    }
    async getScenariosPlayedByUsername(params) {
        const queryParams = {
            username: params.username,
            page: params.page || 0,
            max: params.max || 10
        };
        if (params.sort) {
            queryParams['sort_param[]'] = params.sort;
        }
        return this.request({
            method: 'GET',
            url: this.buildUrl(`/webapp-backend/user/scenario/total-play`, queryParams)
        });
    }
    async getRecentHighScoresByUsername(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl(`/webapp-backend/user/activity/recent`, { username: params.username })
        });
    }
    async getFavoriteScenariosByUsername(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl(`/webapp-backend/user/favorite/scenarios`, { username: params.username })
        });
    }
    async getTotalScenariosCount() {
        return this.request({
            method: 'GET',
            url: '/backend/webapp/custom-scenarios-count'
        });
    }
    async getProfileByUsername(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl(`/webapp-backend/user/profile/by-username`, { username: params.username })
        });
    }
    async getTrendingScenarios() {
        return this.request({
            method: 'GET',
            url: '/webapp-backend/scenario/trending'
        });
    }
    async getMonthlyPlayersCount() {
        return this.request({
            method: 'GET',
            url: '/webapp-backend/user/monthly-players'
        });
    }
    async getConcurrentUsers() {
        return this.request({
            method: 'GET',
            url: '/backend/webapp/concurrent-users'
        });
    }
    async getFeaturedHighScores(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/leaderboard/scores/vip/recent', params)
        });
    }
    async getScenarioDetails(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/scenario/details', { leaderboardId: params.leaderboardId })
        });
    }
    async searchScenarioLeaderboard(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/leaderboard/scores/global', {
                leaderboardId: params.leaderboardId,
                page: params.page,
                max: params.max
            })
        });
    }
    async getLastScoresByScenarioName(params) {
        return this.request({
            method: 'GET',
            url: this.buildUrl('/webapp-backend/user/scenario/last-scores/by-name', {
                username: params.username,
                scenarioName: params.scenarioName
            })
        });
    }
}
exports.KovaaksApiClient = KovaaksApiClient;
