// kovaaks-api-client/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScenarioLeaderboardScoreSearch = exports.FeaturedHighScores = exports.GetScenariosPlayedByUsernameSortedByPlays = exports.GetBenchmarkProgressForWebappUsername = exports.SearchScenariosByScenarioName = void 0;
// kovaaks api tested/SEARCH_SCENARIOS_BY_SCENARIO_NAME.ts
var SearchScenariosByScenarioName;
(function (SearchScenariosByScenarioName) {
    let AimType;
    (function (AimType) {
        AimType["Clicking"] = "Clicking";
        AimType["TargetSwitching"] = "Target Switching";
        AimType["Tracking"] = "Tracking";
    })(AimType = SearchScenariosByScenarioName.AimType || (SearchScenariosByScenarioName.AimType = {}));
})(SearchScenariosByScenarioName || (exports.SearchScenariosByScenarioName = SearchScenariosByScenarioName = {}));
// kovaaks api tested/GET BENCHMARK PROGRESS FOR WEBAPP USERNAME.ts
var GetBenchmarkProgressForWebappUsername;
(function (GetBenchmarkProgressForWebappUsername) {
    let RankName;
    (function (RankName) {
        RankName["DiamondIII"] = "Diamond III";
        RankName["NoRank"] = "No Rank";
        RankName["The05"] = "0/5";
    })(RankName = GetBenchmarkProgressForWebappUsername.RankName || (GetBenchmarkProgressForWebappUsername.RankName = {}));
    let Type;
    (function (Type) {
        Type["Benchmark"] = "benchmark";
        Type["Workout"] = "workout";
    })(Type = GetBenchmarkProgressForWebappUsername.Type || (GetBenchmarkProgressForWebappUsername.Type = {}));
})(GetBenchmarkProgressForWebappUsername || (exports.GetBenchmarkProgressForWebappUsername = GetBenchmarkProgressForWebappUsername = {}));
// kovaaks api tested/GET_SCENARIOS_PLAYED_BY_USERNAME_SORTED_BY_PLAYS.ts
var GetScenariosPlayedByUsernameSortedByPlays;
(function (GetScenariosPlayedByUsernameSortedByPlays) {
    let FovScale;
    (function (FovScale) {
        FovScale["Horizontal43"] = "Horizontal (4:3)";
        FovScale["Overwatch"] = "Overwatch";
    })(FovScale = GetScenariosPlayedByUsernameSortedByPlays.FovScale || (GetScenariosPlayedByUsernameSortedByPlays.FovScale = {}));
    let Resolution;
    (function (Resolution) {
        Resolution["The1600X900"] = "1600x900";
        Resolution["The1920X1080"] = "1920x1080";
    })(Resolution = GetScenariosPlayedByUsernameSortedByPlays.Resolution || (GetScenariosPlayedByUsernameSortedByPlays.Resolution = {}));
    let SensScale;
    (function (SensScale) {
        SensScale["CM360"] = "cm/360";
        SensScale["Overwatch"] = "Overwatch";
        SensScale["QuakeSource"] = "Quake/Source";
    })(SensScale = GetScenariosPlayedByUsernameSortedByPlays.SensScale || (GetScenariosPlayedByUsernameSortedByPlays.SensScale = {}));
})(GetScenariosPlayedByUsernameSortedByPlays || (exports.GetScenariosPlayedByUsernameSortedByPlays = GetScenariosPlayedByUsernameSortedByPlays = {}));
// kovaaks api tested/FEATURED_HIGH_SCORES.ts
var FeaturedHighScores;
(function (FeaturedHighScores) {
    let ClientBuildVersion;
    (function (ClientBuildVersion) {
        ClientBuildVersion["The37720250421102618Ebda1827461E"] = "3.7.7.2025-04-21-10-26-18-ebda1827461e";
    })(ClientBuildVersion = FeaturedHighScores.ClientBuildVersion || (FeaturedHighScores.ClientBuildVersion = {}));
    let Scale;
    (function (Scale) {
        Scale["CM360"] = "cm/360";
        Scale["CallOfDuty"] = "Call of Duty";
        Scale["Overwatch"] = "Overwatch";
        Scale["Valorant"] = "Valorant";
    })(Scale = FeaturedHighScores.Scale || (FeaturedHighScores.Scale = {}));
    let Model;
    (function (Model) {
        Model["None"] = "None";
    })(Model = FeaturedHighScores.Model || (FeaturedHighScores.Model = {}));
    let Resolution;
    (function (Resolution) {
        Resolution["The1600X900"] = "1600x900";
        Resolution["The1920X1080"] = "1920x1080";
        Resolution["The2560X1440"] = "2560x1440";
    })(Resolution = FeaturedHighScores.Resolution || (FeaturedHighScores.Resolution = {}));
    let Game;
    (function (Game) {
        Game["Aimerz"] = "aimerz";
        Game["Cs2"] = "CS2";
        Game["KovaaKS"] = "KovaaK's";
        Game["KovaaKs"] = "KovaaKs";
        Game["Valorant"] = "Valorant";
    })(Game = FeaturedHighScores.Game || (FeaturedHighScores.Game = {}));
})(FeaturedHighScores || (exports.FeaturedHighScores = FeaturedHighScores = {}));
// kovaaks api tested/SCENARIO LEADERBOARD SCORE SEARCH.ts
var ScenarioLeaderboardScoreSearch;
(function (ScenarioLeaderboardScoreSearch) {
    let Hash;
    (function (Hash) {
        Hash["The18505Db399Aaa604B7A65Fa03Ffa7490"] = "18505db399aaa604b7a65fa03ffa7490";
    })(Hash = ScenarioLeaderboardScoreSearch.Hash || (ScenarioLeaderboardScoreSearch.Hash = {}));
    let Model;
    (function (Model) {
        Model["None"] = "None";
    })(Model = ScenarioLeaderboardScoreSearch.Model || (ScenarioLeaderboardScoreSearch.Model = {}));
    let Resolution;
    (function (Resolution) {
        Resolution["The1280X960"] = "1280x960";
        Resolution["The1600X900"] = "1600x900";
        Resolution["The1680X1050"] = "1680x1050";
        Resolution["The1920X1080"] = "1920x1080";
        Resolution["The2560X1440"] = "2560x1440";
    })(Resolution = ScenarioLeaderboardScoreSearch.Resolution || (ScenarioLeaderboardScoreSearch.Resolution = {}));
})(ScenarioLeaderboardScoreSearch || (exports.ScenarioLeaderboardScoreSearch = ScenarioLeaderboardScoreSearch = {}));
